ddtrace/internal/telemetry/telemetry_writer.py

```diff
@@ -0,0 +1,184 @@
+            requests = self._requests_queue
+            self._requests_queue = []
+        return requests
+
+    def periodic(self):
+        integrations = self.flush_integrations_queue()
+        if integrations:
+            integrations_request = app_integrations_changed_telemetry_request(integrations)
+            TelemetryWriter.add_request(integrations_request)
+
+        requests = self.flush_requests_queue()
+        requests_failed = []  # type: List[TelemetryRequest]
+        for request in requests:
+            resp = self._send_request(request)
+            if resp.status >= 300:
+                requests_failed.append(request)
+
+        with self._lock:
+            # add failed requests back to the requests queue
+            self._requests_queue.extend(requests_failed)
+
+    def shutdown(self):
+        # type: () -> None
+        appclosed_request = app_closed_telemetry_request()
+        self.add_request(appclosed_request)
+        self.periodic()
+
+    @classmethod
+    def add_request(cls, request):
+        # type: (TelemetryRequest) -> None
+        """
+        Adds a Telemetry Request to the TelemetryWriter request buffer
+
+         :param TelemetryRequest request: dictionary which stores a formatted telemetry request body and header
+        """
+        with cls._lock:
+            if cls._instance is None:
+                return
+            request["body"]["seq_id"] = cls.sequence
+            cls.sequence += 1
+            # TO DO: replace list with a dead letter queue
+            cls._instance._requests_queue.append(request)
+
+    @classmethod
+    def integration_event(cls, integration_name):
+        # type: (str) -> None
+        """
+        Creates and queues a the name and configurations of a patched module
+
+        :param str integration_name: name of patched module
+        """
+        with cls._lock:
+            if cls._instance is None:
+                return
+            integration = create_integration(integration_name)
+            cls._instance._integrations_queue.append(integration)
+
+    @classmethod
+    def _restart(cls):
+        cls.disable()
+        cls.enable()
+
+    @classmethod
+    def disable(cls):
+        # type: () -> None
+        with cls._lock:
+            if cls._instance is None:
+                return
+
+            forksafe.unregister(cls._restart)
+
+            cls._instance.stop()
+            cls._instance.join()
+            cls._instance = None
+            cls.enabled = False
+
+    @classmethod
+    def enable(cls, endpoint=DEFAULT_TELEMETRY_ENDPOINT):
+        # type: (str) -> None
+        with cls._lock:
+            if cls._instance is not None:
+                return
+
+            telemetry_writer = cls(endpoint)
+            telemetry_writer.start()
+
+            forksafe.register(cls._restart)
+
+            cls._instance = telemetry_writer
+            cls.enabled = True
+
+        request = app_started_telemetry_request()
```

Comment:
can we add a comment here on why we do it here?

we should also add a class method for creating an app started event like integrations

Reply:
I put this in a class method